test/BreadcrumbItemSpec.tsx

```diff
@@ -0,0 +1,165 @@
+    );
+
+    queryAllByRole('listitem', { current: 'page' }).length.should.equal(1);
+  });
+
+  it('Should spread additional props onto inner element', () => {
+    const handleClick = sinon.spy();
+
+    const { getByRole } = render(
+      <Breadcrumb.Item href="#" onClick={handleClick}>
+        Crumb
+      </Breadcrumb.Item>,
+    );
+
+    fireEvent.click(getByRole('button'));
+
+    handleClick.should.have.been.calledOnce;
+  });
+
+  it('Should apply id onto the li element', () => {
+    const { container } = render(
+      <Breadcrumb.Item href="#" id="test-link-id">
+        Crumb
+      </Breadcrumb.Item>,
+    );
+
+    container.querySelectorAll('#test-link-id').length.should.equal(1);
+  });
+
+  it('Should apply `href` property onto `a` inner element', () => {
+    const { getByRole } = render(
+      <Breadcrumb.Item href="http://getbootstrap.com/components/#breadcrumbs">
+        Crumb
+      </Breadcrumb.Item>,
+    );
+
+    const href = getByRole('link').getAttribute('href') || '';
+    href.should.equal('http://getbootstrap.com/components/#breadcrumbs');
+  });
+
+  it('Should apply `title` property onto `a` inner element', () => {
+    const { getByTitle } = render(
+      <Breadcrumb.Item
+        title="test-title"
+        href="http://getbootstrap.com/components/#breadcrumbs"
+      >
+        Crumb
+      </Breadcrumb.Item>,
+    );
+
+    getByTitle('test-title').should.exist;
+  });
+
+  it('Should not apply properties for inner `anchor` onto `li` wrapper element', () => {
+    const { container } = render(
+      <Breadcrumb.Item title="test-title" href="/hi" data-testid>
+        Crumb
+      </Breadcrumb.Item>,
+    );
+
+    container.querySelectorAll('li[href="/hi"]').length.should.equal(0);
+    container.querySelectorAll('li[title="test-title"]').length.should.equal(0);
+  });
+
+  it('Should set `target` attribute on `anchor`', () => {
+    const { getByRole } = render(
+      <Breadcrumb.Item
+        target="_blank"
+        href="http://getbootstrap.com/components/#breadcrumbs"
+      >
+        Crumb
+      </Breadcrumb.Item>,
+    );
+    (getByRole('link').getAttribute('target') as string).should.be.equal(
+      '_blank',
+    );
+  });
+
+  it('Should have li as default component', () => {
+    const { getByTestId } = render(<Breadcrumb.Item data-testid="test" />);
+
+    getByTestId('test').tagName.toLowerCase().should.equal('li');
+  });
+
+  it('Should be able to customize inner link element', () => {
+    const { container } = render(<Breadcrumb.Item linkAs={Button} />);
+
+    container.querySelectorAll('a').length.should.equal(0);
+    container.querySelectorAll('button').length.should.equal(1);
+  });
+
+  it('Should be able to pass props to the customized inner link element', () => {
+    const { getByRole } = render(
+      <Breadcrumb.Item linkAs={Button} linkProps={{ type: 'submit' }} />,
+    );
+
+    (getByRole('button').getAttribute('type') as string).should.equal('submit');
```

Comment:
Is this cast needed?  `getAttribute` should already be a string?

Reply:
The signature of `getAttribute` returns a `string | null` so I added a casting to avoid `Object is possibly 'null'.ts(2531)` error.

Reply:
Another thing you could do is use a non-null assertion:

https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator

`expect` works just as well 😄